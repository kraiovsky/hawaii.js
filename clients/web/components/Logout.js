import React from 'react'
import Link from 'next/link'
import { useGlobalState, logout } from '../store'

export default () => {
  const [, dispatch] = useGlobalState()
  return (
    <>
      <Link href="#">
        <a onClick={() => logout(dispatch)}>Log out</a>
      </Link>
    </>
  )
}
