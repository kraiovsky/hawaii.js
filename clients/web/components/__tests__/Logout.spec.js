import React from 'react'
import renderer from 'react-test-renderer'
import Logout from '../Logout'

jest.mock('react-redux')

test('Render Logout link for the nav bar', () => {
  const component = renderer.create(<Logout />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
