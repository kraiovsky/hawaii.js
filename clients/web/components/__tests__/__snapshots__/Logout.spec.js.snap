// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Logout link for the nav bar 1`] = `
<a
  href="#"
  onClick={[Function]}
>
  Log out
</a>
`;
